### ISSUE TITLE
add_classname to the main element in editor VB?

### ISSUE THREAD

**Issue by mrkenng** [elegantthemes/create-divi-extension#41]
I was playing alright with add_classname with the divi-extension-example module 

```
$this->add_classname( array(
	'this-is-the-class-i-want-to-add-on-vb',
) );
```
It does render out the css class main DIV  on the frontend. 
<img width="1112" alt="screen shot 2018-04-28 at 6 58 09 pm" src="https://user-images.githubusercontent.com/9147887/39395851-78d06cd6-4b16-11e8-9e7f-67f928266201.png">

How do i add the class to the main DIV. 
<img width="1112" alt="screen shot 2018-04-28 at 6 57 38 pm" src="https://user-images.githubusercontent.com/9147887/39395858-9dc00358-4b16-11e8-9b6a-d353dcc2af21.png">

Thanks.

**Comment by etstaging** [comment]
### QA Checklist
<strong>Reporter</strong>
- [ ] Helpful title used &nbsp;<img src="https://assets-cdn.github.com/images/icons/emoji/unicode/274c.png" width="16" height="16"/>
  * ***Helpful title format:*** `Scope(s) :: One Sentence To Describe The Issue`
  * ***Example:*** `VB :: Floating Toolbar :: Jumps To Top Of Page`
- [ ] Detailed description <img src="https://assets-cdn.github.com/images/icons/emoji/unicode/274c.png" width="16" height="16"/>
  * ***Description should usually be 150+ characters.***
- [ ] Steps to reproduce included <img src="https://assets-cdn.github.com/images/icons/emoji/unicode/274c.png" width="16" height="16"/>
- [ ] Support Ticket URL included
- [ ] Image and/or gif

<strong>Developer</strong>
- [ ] Steps to reproduce proven solved
- [ ] Test included and coverage acceptable and all Tests passing?
- [ ] PR description has layman's description of problem and solution and "Areas Affected" listed/explained?
- [ ] PR branch includes latest commits from master branch (at the time of checking)?
- [ ] Estimated points on issue
- [ ] Proper code documentation and conventions?
- [ ] Proper RTL appearance
- [ ] i18n used if applicable
- [ ] Ajax/Form used? Nonce used? Perms checked? All user input sanitized and validated? Intentional unsanitized/unescaped functions used?
- [ ] Database saved/updated? Data sanitized and validated?
- [ ] Variable output properly escaped and validated?

<strong>Peer Review</strong>
- [ ] PR branch includes latest commits from master branch (at the time of checking)?
- [ ] Steps to reproduce proven solved
- [ ] Affected areas are tested and ok
- [ ] Proper code documentation and conventions
- [ ] Proper RTL appearance
- [ ] i18n used if applicable
- [ ] Test(s) included and coverage acceptable
- [ ] Ajax/Form used? Nonce used? Perms checked? All user input sanitized and validated? Intentional unsanitized/unescaped functions used?
- [ ] Database saved/updated? Data sanitized and validated?
- [ ] Variable output properly escaped and validated?

<strong>Quality Assurance</strong>
- [ ] All tests passed on CircleCI.
- [ ] Steps to reproduce passed, the fix works.
- [ ] No errors in console or output 
- [ ] Affected areas are tested and ok
- [ ] The fix does not change a customer's website in any surprising or unintended way
  - [ ] If Divi's design <em>is</em> adjusted, settings are migrated gracefully and invisibly

<strong>Design Review</strong>
- [ ] UI is to spec
- [ ] UX is to spec
- [ ] Design is to spec
- [ ] No surprising visual changes to a customer's website will be experienced after updating
  - [ ] Any/all visual changes are migrated gracefully

<strong>Dev Review</strong>
- [ ] Test(s) included and coverage acceptable
- [ ] Ajax/Form used? Nonce used? Perms checked? All user input sanitized and validated? Intentional unsanitized/unescaped functions used?
- [ ] Database saved/updated? Data sanitized and validated?
- [ ] Variable output properly escaped and validated? 
- [ ] i18n used if applicable
- [ ] Implementation acceptable, DRY, Scaleable, future safe
- [ ] Best practices used, cleverness and grokage, ES6 use, etc.

**Comment by mrkenng** [comment]
Thanks for your reply. However I'm not referring to declaration of CSS style. I think my question was edited. 

Let me rephrase, On the php I am using add_classname so that it can the class name appear on the main element div on the frontend. However it is not in the VB and how do I add a class name in the VB. Using the same method. To the main element div where the class name et_pb_mobule is.

**Comment by xxtesaxx** [comment]
I understood that. The question I ask myself though is why it is absolutely necessary to add a class to the main module. Is it instead possible to wrap your output in a div like your php output could be:

    return '<div class="my-special-class>' . $output . '</div>;

and your React components output could be:

    return (<div className="my-special-class">{output}</div>);

This way it wouldn't matter that you can't add the class to the React components module wrapper. :) 

It's not exactly what you asked for but it would allow you to do whatever you need to do to apply the same CSS in both cases.

**Comment by jamesfosker** [comment]
@xxtesaxx @lots0logs neither of these options are helpful when it comes to the outermost div have dynamic CSS classes depending on what features the user turns on

**Comment by xxtesaxx** [comment]
I think you are missing the bigger picture. There is really no need to add any custom classes yourself to the outermost wrapper. Simply return a "<div class="my-module-wrapper">{actual content}</div>" inside your render function. You then can address any element using css as usual. Divis default selectors can all be modified to target your own wrapper as well (e.g. box shadow, border, background etc) but still I don't see any need for this. 

How about you tell us what you try to achieve so we can better understand what you are having problems with. :)
